smart-contract-verifier/sig-provider-extension/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use http::Uri;
+use serde::Deserialize;
+use serde_with::{serde_as, DisplayFromStr};
+use sig_provider_proto::blockscout::sig_provider::v1::{
+    signature_service_client::SignatureServiceClient, CreateSignaturesRequest,
+};
+use smart_contract_verifier::{Middleware, SourcifySuccess, VerificationSuccess};
+use std::sync::Arc;
+use tonic::transport::Channel;
+
+#[serde_as]
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    #[serde_as(as = "DisplayFromStr")]
+    url: Uri,
+}
+
+pub struct SigProvider {
+    inner: Arc<SigProviderImpl>,
+}
+
+impl SigProvider {
+    pub async fn new(config: Config) -> Result<Self, tonic::transport::Error> {
+        let connection = Channel::builder(config.url).connect().await?;
+        Ok(Self {
+            inner: Arc::new(SigProviderImpl { connection }),
+        })
+    }
+}
+
+struct SigProviderImpl {
+    connection: Channel,
+}
+
+impl SigProviderImpl {
+    async fn create_signatures(self: Arc<Self>, abi: String) {
```

Comment:
Why not a usual:
```suggestion
    async fn create_signatures(&self, abi: String) {
```
?